### PR TITLE
Optimize the latency of lat-sig-install

### PR DESCRIPTION
**Pull request by StevenJiang1110**
This PR optimizes the latency of lat-sig-install from 0.21us to 0.19us(About 9%). The main job is avoiding cloning `VmSpace` when creating `CurrentUserSpace`. 

But this PR actually breaks the abstraction of `CurrentUserSpace`, since after this PR, `CurrentUserSpace` can be built from a not current task. I'm not very sure if this PR is a suitable change.

### CODE REVIEW

File: kernel/src/context.rs [hunk 1]
```diff
@@ -27,30 +27,29 @@ pub struct Context<'a> {
...
-    /// If you get the access to the [`Context`].
-    pub fn get() -> Self {
-        let vm_space = {
-            let current_task = Task::current().unwrap();
-            let user_space = current_task.user_space().unwrap();
-            user_space.vm_space().clone()
-        };
-        Self(vm_space)
+impl<'a> CurrentUserSpace<'a> {
+    pub fn new(task: &'a Task) -> Self {
```
Comment (tatetian) [inline kernel/src/context.rs:51]: Although we cannot prevent users from calling this constructor,  we should at least make it harder to misuse.

```rust
impl<'a> CurrentUserSpace<'a> {
    /// This constructor method is for internal use only.
    ///
    /// Use the `get_current_userspace` macro to get an instance of `CurrentUserSpace`.
    /// If you already have access to `Context`, call `Context::get_user_space`.
    #[doc(hidden)]
    pub fn _new(task: &'a Task) -> Self {
```

**Review by tatetian — COMMENTED** [review]
The PR looks mostly good. Just one tiny issue.

File: kernel/src/context.rs (hunk 1 above)
Comment (StevenJiang1110) [reply kernel/src/context.rs:64]: I have refined the doc and added a debug assert to make this method hard to misuse.

**Review by tatetian — APPROVED** [review]
LGTM. Thanks for the contribution!

**Pull request merged**